app/assets/javascripts/tylium/modules/auto_save/local.js

```diff
@@ -0,0 +1,127 @@
+class LocalAutoSave {
+  constructor(target) {
+    if (target.tagName !== 'FORM') { console.error('Can\'t initialize local auto save on anything but a form'); return; }
+    this.target = target;
+    this.key = target.dataset.autoSaveKey;
+    this.cancelled = false;
+    this.submitted = false;
+
+    // Don't store authenticity_token and utf8
+    this.excludedInputNames = ['utf8', 'authenticity_token', '_method'];
+
+    this.init();
+  }
+
+  init() {
+    this.behaviors();
+    this.restoreData();
+  }
+
+  behaviors() {
+    var that = this;
+
+    this.target.addEventListener('submit', function(event) {
+      that.submitted = true;
+      localStorage.removeItem(that.key);
+    });
+
+    var clearCacheElement = this.target.querySelector('[data-behavior~=clear-local-auto-save]');
+
+    if (clearCacheElement) {
+      clearCacheElement.addEventListener('click', function(event) {
+        that.cancelled = true;
+        localStorage.removeItem(that.key);
+      })
+    }
+
+    this._debounceTimer = 500;
+
+    this.target.querySelectorAll('input, textarea, select').forEach(function(input) {
+      // Don't add event handler for submit button
+      if (input.type === 'button') { return true; }
+
+      // we're using a jQuery plugin for :textchange event, so need to use $()
+      $(input).on('textchange change', that.handleTextChange.bind(that));
+    })
+  }
+
+  handleTextChange() {
+    clearTimeout(this._debounceTimer);
```

Comment:
I've double used a variable in this example and shouldn't have. 

It should read:

```javascript
 handleTextChange() {
    clearTimeout(this._debounceTimeout);

    this._debounceTimeout = setTimeout(function() {
      this.setData();
    }.bind(this), this._debounceTimer);
  }
```

`this._debounceTimer` should represent the timing interval and `this._debounceTimeout` should be the timeout. One was clobbering the other.